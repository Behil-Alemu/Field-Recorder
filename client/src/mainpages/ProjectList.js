import React, { useContext } from 'react';
//import FoldersApi from '../api/FoldersApi';
import UserContext from '../auth/UserContext';
import LoadingSpinner from '../helpers/LoadingSpinner';
import { Card, Text, CardBody, CardHeader, Heading, Box, SimpleGrid, useColorModeValue } from '@chakra-ui/react';
import ProjectCard from './ProjectCard';

function ProjectList({ projects, setProjects }) {
	// const [ formErrors, setFormErrors ] = useState([]);
	const { currentUser } = useContext(UserContext);
	const bg = useColorModeValue('gray.100', 'gray.600');
	const cardBg = useColorModeValue('red.50', 'red.800');

	console.debug('ProjectList', 'currentUser=', currentUser, 'projects=', projects);

	if (!projects) {
		console.log('projects not hereee');
		return <LoadingSpinner />;
	}
	const updateProjects = (deletedProjectId) => {
		setProjects((prevProjects) => prevProjects.filter((project) => project.id !== deletedProjectId));
	};

	return (
		<div>
			{projects.length ? (
				<Box p={4} rounded="md" bg={bg}>
					<SimpleGrid spacing={4} templateColumns="repeat(auto-fill, minmax(200px, 1fr))">
						{projects.map((p) => (
							<ProjectCard
								key={p.id}
								id={p.id}
								folderName={p.foldername}
								updateProjects={updateProjects}
							/>
						))}
					</SimpleGrid>
				</Box>
			) : (
				<Card maxW="md" align="center" bg={cardBg}>
					<CardHeader>
						<Heading size="md">Sorry, no projects yet!</Heading>
					</CardHeader>
					<CardBody>
						<Text>Use above form :|</Text>
					</CardBody>
				</Card>
			)}
		</div>
	);
}

export default ProjectList;
