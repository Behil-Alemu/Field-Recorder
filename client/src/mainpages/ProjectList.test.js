import React from 'react';
import { render } from '@testing-library/react';
import ProjectList from './ProjectList';
import { MemoryRouter } from 'react-router';
import { UserProvider } from '../testUtils';

it('matches snapshot', function() {
	const { asFragment } = render(
		<MemoryRouter>
			<UserProvider>
				<ProjectList currentUser={null} />
			</UserProvider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
