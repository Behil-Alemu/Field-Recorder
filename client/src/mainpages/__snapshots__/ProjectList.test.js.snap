// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-spinner css-1oq7xc3"
  >
    <span
      class="css-8b45rq"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
